homeassistant/components/homekit/accessories.py

```diff
@@ -120,8 +130,16 @@ def update_battery(self, new_state):
 
         Only call this function if self._support_battery_level is True.
         """
-        battery_level = convert_to_float(
-            new_state.attributes.get(ATTR_BATTERY_LEVEL))
```

Comment:
You should be able to revert the changes to the `update_battery` method without any functionality missing.

Reply:
I had to slightly modify `update_battery` since battery value is the `state` property on `new_state`, unlike the previous implementation which uses `attributes`.